packages/route/src/Domain/Model/Route.php

```diff
@@ -45,6 +45,13 @@ public function __construct(string $resourceKey, string $resourceId, string $loc
 
     public function setSlug(string $slug): static
     {
+        if ($this->parentRoute
+            && $this->slug !== $slug
+            && static::extractParentSlug($slug) !== static::extractParentSlug($this->slug)
+        ) {
+            $this->parentRoute = null;
```

Comment:
lets discuss this deeper with @chirimoya. Technically we don't need remove the relation, which maybe, it only depends on what should happening if we do:

```
 - parent
    - parent/child
        - parent/child/grand-child -> parent/grand-child
```

->

```
 - parent -> changed
     - parent/child
        - parent/grand-child
```

what is the expected result:

```
 - changed
    - changed/child
        - parent/grand-child
```

or

```
 - changed
    - changed/child
        - changed/grand-child
```

Reply:
implemented now